Mage.Sets/src/mage/cards/s/ScarwoodBandits.java

```diff
@@ -0,0 +1,191 @@
+                new GenericManaCost(2)),
+            new ManaCostsImpl("{2}{G}"));
+        ability.addCost(new TapSourceCost());
+        ability.addTarget(new TargetArtifactPermanent());
+        this.addAbility(ability);
+    }
+
+    public ScarwoodBandits(final ScarwoodBandits card) {
+        super(card);
+    }
+
+    @Override
+    public ScarwoodBandits copy() {
+        return new ScarwoodBandits(this);
+    }
+}
+
+class DoUnlessAnyOpponentPaysEffect extends OneShotEffect {
+
+    protected Effects executingEffects = new Effects();
+    private final Cost cost;
+    private String chooseUseText;
+
+    public DoUnlessAnyOpponentPaysEffect(Effect effect, Cost cost) {
+        this(effect, cost, null);
+    }
+
+    public DoUnlessAnyOpponentPaysEffect(Effect effect, Cost cost, String chooseUseText) {
+        super(Outcome.Benefit);
+        this.executingEffects.add(effect);
+        this.cost = cost;
+        this.chooseUseText = chooseUseText;
+    }
+
+    public DoUnlessAnyOpponentPaysEffect(final DoUnlessAnyOpponentPaysEffect effect) {
+        super(effect);
+        this.executingEffects = effect.executingEffects.copy();
+        this.cost = effect.cost.copy();
+        this.chooseUseText = effect.chooseUseText;
+    }
+
+    public void addEffect(Effect effect) {
+        executingEffects.add(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        MageObject sourceObject = game.getObject(source.getSourceId());
+        if (controller != null && sourceObject != null) {
+            String message;
+            if (chooseUseText == null) {
+                String effectText = executingEffects.getText(source.getModes().getMode());
+                message = "Pay " + cost.getText() + " to prevent (" + effectText.substring(0, effectText.length() - 1) + ")?";
+            } else {
+                message = chooseUseText;
+            }
+            message = CardUtil.replaceSourceName(message, sourceObject.getName());
+            boolean result = true;
+            boolean doEffect = true;
+            // check if any opponent is willing to pay
+            for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
+                Player player = game.getPlayer(playerId);
+                if (player != null && player != controller && cost.canPay(source, source.getSourceId(), player.getId(), game) && player.chooseUse(Outcome.Detriment, message, source, game)) {
+                    cost.clearPaid();
+                    if (cost.pay(source, game, source.getSourceId(), player.getId(), false, null)) {
+                        if (!game.isSimulation()) {
+                            game.informPlayers(player.getLogName() + " pays the cost to prevent the effect");
+                        }
+                        doEffect = false;
+                        break;
```

Comment:
No break needed here, people still get the option to pay even if nobody paid. I know, it's weird.